src/qibolab/instruments/zhinst.py

```diff
@@ -940,8 +980,9 @@ def sweep_recursion(self, qubits, exp, exp_calib, exp_options):
                 )
         if sweeper.parameter is Parameter.amplitude:
             for pulse in sweeper.pulses:
-                pulse.amplitude *= max(sweeper.values)
-                sweeper.values /= max(sweeper.values)
+                sweeper.values = sweeper.values.copy()
```

Comment:
Why do you copy the values only?

You have access to these values only through `sweeper` (it seems to me), so overwriting with a copy it is just replacing the original with a copy of itself, but you are not keeping the original anywhere...